config/config-mp/src/main/java/io/helidon/config/mp/MpMetaConfig.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.config.mp;
+
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import io.helidon.config.Config;
+import io.helidon.config.ConfigSources;
+import io.helidon.config.spi.ConfigSource;
+
+final class MpMetaConfig {
+    private static final String META_CONFIG_ENV_VAR = "HELIDON_MP_META_CONFIG";
+    static final String META_CONFIG_SYSTEM_PROPERTY = "io.helidon.config.mp.meta-config";
+
+    private static final Logger LOGGER = Logger.getLogger(MpMetaConfig.class.getName());
+
+    private MpMetaConfig() {
+    }
+
+    static Optional<Config> metaConfig() {
+        return findMetaConfig()
+                .map(it -> Config.builder()
+                        .disableValueResolving()
+                        .disableKeyResolving()
+                        .disableFilterServices()
+                        .disableEnvironmentVariablesSource()
+                        .disableSystemPropertiesSource()
+                        .disableMapperServices()
+                        .addSource(it)
+                        .build());
+    }
+
+    private static Optional<ConfigSource> findMetaConfig() {
+        String fileName = System.getenv(META_CONFIG_ENV_VAR);
+        if (fileName == null) {
+            fileName = System.getProperty(META_CONFIG_SYSTEM_PROPERTY);
+        }
+        Optional<ConfigSource> found;
+        if (fileName == null) {
+            found = Optional.empty();
+        } else {
+            found = findMetaConfig(fileName);
+        }
+
+        return found.or(() -> findMetaConfig("mp-meta-config.yaml"))
+                .or(() -> findMetaConfig("mp-meta-config.properties"));
+    }
```

Comment:
Could this do the same?
```java
        return Optional.ofNullable(System.getenv(META_CONFIG_ENV_VAR))
                .or(() -> Optional.ofNullable(System.getProperty(META_CONFIG_SYSTEM_PROPERTY)))
                .flatMap(Scratch::findMetaConfig)
                .or(() -> findMetaConfig("mp-meta-config.yaml"))
                .or(() -> findMetaConfig("mp-meta-config.properties"));
```

Reply:
I had it there, but I do not like the `Optional.ofNullable` used twice - it seems like an overhead just to have a flow API used..

Reply:
np